Mathlib/RingTheory/ZMod/UnitsCyclic.lean

```diff
@@ -0,0 +1,520 @@
+/-
```

Comment:
My golfing of this file is in [this commit](https://github.com/leanprover-community/mathlib4/commit/07a9ccabbd6a6ac8188eed4cd40627876a77ace9); `isCyclic_units_four_mul_iff` and the final theorem remain to be golfed.